### PR TITLE
builtin: impl sec_to_time

### PR DESCRIPTION
**Pull request by EndlessIdea**
Hi，I added the sec_to_time function. Please help review it. Thanks very much.

### CODE REVIEW

**Comment by shenli** [comment]
@EndlessIdea Thanks for your PR! Please sign the CLA and resolve the conflicts.

**Comment by coocood** [comment]
@EndlessIdea 
Thank you for your PR.
The user email in your git user commit is not the same as your Github account.
Please set the email in your git config with your Github email.

**Comment by EndlessIdea** [comment]
Thank you. I've added the commit email to my accout,and sign the cla.

File: expression/builtin_time.go [hunk 1]
```diff
@@ -1914,7 +1914,47 @@ type builtinSecToTimeSig struct {
...
+
+	var (
+		hour     int64
+		minute   int64
+		second   int64
+		negative string
+	)
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+	seconds, _ := args[0].ToInt64(sc)
```
Comment (coocood) [inline expression/builtin_time.go]: Please handle the error.

File: expression/builtin_time_test.go [hunk 2]
```diff
@@ -1248,3 +1249,27 @@ func (s *testEvaluatorSuite) TestTimeToSec(c *C) {
...
+	// Test cases from https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_sec-to-time
+	tests := []struct {
+		param  interface{}
+		expect string
+	}{
+		{2378, "00:39:38"},
+		{3864000, "838:59:59"},
+		{-3864000, "-838:59:59"},
+		{86401.4, "24:00:01"},
+		{86401.5, "24:00:02"},
```
Comment (coocood) [inline expression/builtin_time_test.go]: Please add cases for `nil` and valid string like "123" and  invalid string like "abc".

Comment (coocood) [inline expression/builtin_time_test.go]: MySQL returnes result with fraction number `24:00:02.5`.

Comment (EndlessIdea) [inline expression/builtin_time_test.go]: ok,I'll add the cases.
But the result shows like this, when I use "select sec_to_time(23.5);"
mysql> select sec_to_time(23.5
);
+-------------------+
| sec_to_time(23.5) |
+-------------------+
| 00:00:24          |
+-------------------+
1 row in set
the result doesn't has faction number.
Is my mysql version too old?

Comment (coocood) [inline expression/builtin_time_test.go]: It has fraction result on 5.6.38 and 5.7.17, which version is yours?

Comment (EndlessIdea) [inline expression/builtin_time_test.go]: When I use a invalid string like "abc", the result shows like below.
mysql> select sec_to_time("abc");
+--------------------+
| sec_to_time("abc") |
+--------------------+
| 00:00:00           |
+--------------------+
1 row in set
So do I treat it as a invalid string to return error or just return "00:00:00"?
Still has another question:
How do I add the nil case? I don't know what the expect param should be.
Thank you!

Comment (EndlessIdea) [inline expression/builtin_time_test.go]: My version is 5.5.40. OK, I get it.

Comment (coocood) [inline expression/builtin_time_test.go]: 1. it is controlled by `sql_mode` and the query type, which is handled by `StatementContext`, you don't need to worry about.
2. just use `nil` as param will do.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM
Please resolve conflict. @EndlessIdea

**Comment by EndlessIdea** [comment]
@coocood @tiancaiamao Thank you. I'm done.

**Pull request merged**